src/plugins/data/server/search/routes/msearch.ts

```diff
@@ -93,35 +59,15 @@ export function registerMsearchRoute(router: IRouter, deps: SearchRouteDependenc
       },
     },
     async (context, request, res) => {
-      const client = context.core.elasticsearch.client.asCurrentUser;
-
-      // get shardTimeout
-      const config = await deps.globalConfig$.pipe(first()).toPromise();
-      const timeout = getShardTimeout(config);
-
-      const body = convertRequestBody(request.body, timeout);
-
-      // trackTotalHits is not supported by msearch
-      const { trackTotalHits, ...defaultParams } = await getDefaultSearchParams(
-        context.core.uiSettings.client
-      );
+      const callMsearch = getCallMsearch({
+        esClient: context.core.elasticsearch.client,
+        globalConfig$: deps.globalConfig$,
+        uiSettings: context.core.uiSettings.client,
+      });
```

Comment:
This got simplified to a helper which could also be called outside of a route handler